specification/eventgrid/data-plane/Microsoft.Communication/stable/2018-01-01/AzureCommunicationServices.json

```diff
@@ -1960,6 +2197,145 @@
           "format": "date-time"
         }
       }
+    },
+    "AcsAdvancedMessageChannelEventError": {
+      "description": "Advanced Message Channel Event Error",
+      "type": "object",
+      "properties": {
+        "channelCode": {
+          "description": "The channel error code",
+          "type": "string"
+        },
+        "channelMessage": {
+          "description": "The channel error message",
+          "type": "string"
+        }
+      }
+    },
+    "AcsAdvancedMessageMediaContent": {
+      "description": "Advanced Message Media Content",
+      "type": "object",
+      "properties": {
+        "mimeType": {
+          "description": "The MIME type of the file this media represents",
+          "type": "string"
+        },
+        "id": {
+          "description": "The media identifier",
+          "type": "string"
+        },
+        "fileName": {
+          "description": "The filename of the underlying media file as specified when uploaded",
+          "type": "string"
+        },
+        "caption": {
+          "description": "The caption for the media object, if supported and provided",
+          "type": "string"
+        }
+      }
+    },
+    "AcsAdvancedMessageContext": {
+      "description": "Advanced Message Context",
+      "type": "object",
+      "properties": {
+        "from": {
+          "description": "The WhatsApp ID for the customer who replied to an inbound message.",
+          "type": "string"
+        },
+        "id": {
+          "description": "The message ID for the sent message for an inbound reply",
+          "type": "string"
+        }
+      }
+    },
+    "AcsAdvancedMessageButtonContent": {
+      "description": "Advanced Message Button Content",
+      "type": "object",
+      "properties": {
+        "text": {
+          "description": "The Text of the button",
+          "type": "string"
+        },
+        "payload": {
+          "description": "The Payload of the button which was clicked by the user, setup by the business",
+          "type": "string"
+        }
+      }
+    },
+    "AcsAdvancedMessageInteractiveContent": {
+      "description": "Advanced Message Interactive Content",
+      "type": "object",
+      "properties": {
+        "type": {
```

Comment:
This needs an x-ms-client-name of "kind"

Reply:
done